include/z64pause_menu.h

```diff
@@ -0,0 +1,245 @@
+    /* 0x29E */ s16 promptAlpha;
+    /* 0x2A0 */ s16 ocarinaSongIndex;
+    /* 0x2A2 */ u8 worldMapPoints[20];
+    /* 0x2B6 */ u8 unk_2B6; // unused red?
+    /* 0x2B7 */ u8 unk_2B7; // unused green?
+    /* 0x2B8 */ u8 unk_2B8; // unused blue?
+    /* 0x2B9 */ u8 itemDescriptionOn; // helpful description of item given through a message box
+    /* 0x2BA */ s16 equipAnimScale; // scale of item icon while moving being equipped to c-button
+    /* 0x2BC */ s16 equipAnimShrinkRate; // rate the scale is shrinking for the item icon while moving being equipped to c-button
+    /* 0x2BE */ s16 ocarinaButtonsY[5];
+    /* 0x2C8 */ u16 unk_2C8; // Uses PauseMenuPage enum for Owl Warp. Never set.
+    /* 0x2CA */ s16 unk_2CA; // Uses OwlWarpId enum for Owl Warp. Never set.
+} PauseContext; // size = 0x2D0
+
+#define IS_PAUSE_STATE_GAMEOVER \
+    ((pauseCtx->state >= PAUSE_STATE_GAMEOVER_0) && (pauseCtx->state <= PAUSE_STATE_GAMEOVER_10))
+
+#define IS_PAUSE_STATE_OWLWARP \
+    ((pauseCtx->state >= PAUSE_STATE_OWLWARP_2) && (pauseCtx->state <= PAUSE_STATE_OWLWARP_6))
+
+#define IS_PAUSE_MAIN_STATE_SONG_PROMPT                            \
+    ((pauseCtx->mainState >= PAUSE_MAIN_STATE_SONG_PROMPT_INIT) && \
+     (pauseCtx->mainState <= PAUSE_MAIN_STATE_SONG_PROMPT_DONE))
+
+
+typedef enum KaleidoMgrOverlayType {
+    /* 0 */ KALEIDO_OVL_KALEIDO_SCOPE,
+    /* 1 */ KALEIDO_OVL_PLAYER_ACTOR,
+    /* 2 */ KALEIDO_OVL_MAX
+} KaleidoMgrOverlayType;
+
+typedef struct KaleidoMgrOverlay {
+    /* 0x00 */ void* loadedRamAddr;
+    /* 0x04 */ uintptr_t vromStart;
+    /* 0x08 */ uintptr_t vromEnd;
+    /* 0x0C */ void* vramStart;
+    /* 0x10 */ void* vramEnd;
+    /* 0x14 */ uintptr_t offset; // loadedRamAddr - vramStart
+    /* 0x18 */ const char* name;
+} KaleidoMgrOverlay; // size = 0x1C
+
+
+// z_kaleido_setup.c
+
+void func_800F4A10(struct PlayState* play);
+void KaleidoSetup_Update(struct PlayState* play);
+void KaleidoSetup_Init(struct PlayState* play);
+void KaleidoSetup_Destroy(struct PlayState* play);
+
+
+// z_kaleido_manager.c
+
+void KaleidoManager_LoadOvl(KaleidoMgrOverlay* ovl);
+void KaleidoManager_ClearOvl(KaleidoMgrOverlay* ovl);
+void KaleidoManager_Init(struct PlayState* play);
+void KaleidoManager_Destroy(void);
+void* KaleidoManager_GetRamAddr(void* vram);
+
+extern KaleidoMgrOverlay gKaleidoMgrOverlayTable[KALEIDO_OVL_MAX];
+extern KaleidoMgrOverlay* gKaleidoMgrCurOvl;
+
+
+// z_kaleido_scope_call.c
+
+void KaleidoScopeCall_LoadPlayer(void);
+void KaleidoScopeCall_Init(struct PlayState* play);
+void KaleidoScopeCall_Destroy(struct PlayState* play);
+void KaleidoScopeCall_Update(struct PlayState* play);
+void KaleidoScopeCall_Draw(struct PlayState* play);
+
+
+// Item macros
+
+#define ITEM_GRID_ROWS 4
+#define ITEM_GRID_COLS 6
+#define MASK_GRID_ROWS 4
+#define MASK_GRID_COLS 6
+
+#define ITEM_NUM_SLOTS (ITEM_GRID_ROWS * ITEM_GRID_COLS)
+#define MASK_NUM_SLOTS (MASK_GRID_ROWS * MASK_GRID_COLS)
```

Comment:
this feels like it shouldn't be public but I see z_sram uses the `_num_slots` values
(in oot afaict these can be private without problem)

Should/could z_sram use new defines like `SLOT_ITEM_MAX`, `SLOT_MASK_FIRST` instead?

Reply:
z_sram using this value surprised me too.
I'm not sure what would be better to do here. @engineer124 do you have any ideas?

Reply:
This has to do with MM's inventory items grouping items and masks into 1 variable (would be worth double checking if they could split), meaning anytime masks are accessed, it's accessed as `items[24 + slot]`
```c
typedef struct Inventory {
    /* 0x00 */ u8 items[48]; // Contains the 24 slots for items and 24 slots for mask
   ...
```

So I think there should be both a `ITEM_NUM_SLOTS` and `MASK_NUM_SLOTS` for save (it's used more than just outside of `z_sram`, and in theory should stay in sync with the same value that the pause menu uses which is 24 = 4 * 6, so idk. Open to thoughts

Reply:
>would be worth double checking if they could split

From what I can tell, they cannot. In particular `func_801457CC` does not like them being split.